src/components/common/ModalPortal.jsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+function ModalPortal({ children }) {
+  const [container] = useState(() => document.createElement('div'));
+
+  useEffect(() => {
+    const modalRoot = document.getElementById('modal-root');
+
+    if (!modalRoot) {
+      // eslint-disable-next-line no-console
+      console.error('#modal-root element not found in index.html');
+      return;
+    }
+
+    modalRoot.appendChild(container);
+
+    return () => {
+      modalRoot.removeChild(container);
+    };
+  }, [container]);
+
+  return createPortal(children, container);
+}
```

Comment:
지금 useState로 모달 포탈을 위한 div를 생성하는 코드를 작성해주셨는데
index.html에 `<div class="modal-root"></div>`를 이미 추가해주셔서
아래처럼 코드를 리팩토링 해볼 수 있을 것 같습니다!

```suggestion
const ModalPortal = ({ children }) => {
  const modalRoot = document.getElementById('modal-root');

  if (!modalRoot) {
      console.error('#modal-root element not found. Modal will not render.');
      return null; 
  }

  return createPortal(children, modalRoot);
};
```